daffodil-test/src/test/resources/org/apache/daffodil/section23/dfdl_expressions/expressions.tdml

```diff
@@ -27,7 +27,48 @@
   xmlns:daf="urn:ogf:dfdl:2013:imp:daffodil.apache.org:2018:ext"  
   xmlns:fn="http://www.w3.org/2005/xpath-functions"
   defaultRoundTrip="true">
-  
+
+  <tdml:defineSchema name="DFDLCheckRange">
+    <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd"/>
+    <dfdl:format ref="ex:GeneralFormat" />
+    <xs:element name="root1" dfdl:lengthKind="explicit" dfdl:length="1" type="xs:integer">
+      <xs:annotation>
+        <xs:appinfo source="http://www.ogf.org/dfdl/">
+          <dfdl:discriminator test="{ dfdl:checkRangeInclusive(., 0, 1) }" message="Discriminator failed ..." />
```

Comment:
Although discrminators works, it adds extra complexity to the test that can sometimes make it difficult to ensure if the function is behaving correctly. I would suggest just using inputValueCalc, e.g.:

```xml
<xs:element name="isInRange" ... dfdl:inputValueCalc="{ dfdl:checkRangeInclusive(...) }" />
```

This way you can get actual true/false values in the expected infoset, rather than assuming a discriminator passing/failing implies the function worked as expected.